homeworks/src/homework4/part2/TestCase.java

```diff
@@ -0,0 +1,92 @@
+package homework4.part2;
+
+import java.io.*;
+
+public class TestCase {
+    public static void main(String[] args) {
+
+        Obstacle[] goals = {
+                new Runway(200),
+                new Wall(4, 60),
+                new Wall(6, 150),
+                new Runway(400),
+                new Wall(4, 60),
+                new Wall(6, 150),
+                new Wall(8, 300),
+                new Wall(3, 380),
+                new Runway(1000),
+                new Wall(5, 300),
+                new Wall(3, 800),
+                new Runway(1500),
+                new Wall(8, 300),
+                new Wall(2, 600),
+                new Wall(4, 700),
+                new Wall(6, 1200)
+        };
+
+        Participant[] participants = {
+                new Human(300, 5, "Калдырь"),
+                new Human(600, 10, "Бугай"),
+                new Human(150, 8, "Куряга"),
+                new Robot(100, 10, "Робот Дрим"),
+                new Robot(800, 5, "Робот Драм"),
+                new Robot(350, 3, "Робот Друм"),
+                new Cat(1800, 10, "Кот Кузя"),
+                new Robot(1000, 8, "Кошка Муся"),
+                new Robot(200, 7, "Кот Толстопуз"),
+        };
+
+        int distanceDone = 0;
+
+        for (Participant participant : participants) {
+            for (Obstacle goal : goals) {
+                if (goal instanceof Runway) {
```

Comment:
**The main goal was to use polymorphism**, for example:
In class Runway implements overcame(Participant participant){patricipant.run();}

As the result, in every Obstacle, you will call "overcame" only with the needed participant